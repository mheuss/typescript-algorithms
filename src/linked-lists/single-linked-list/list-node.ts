export class ListNode<T> {
  public next: ListNode<T> | null = null;
  public payload: T;

  constructor(payload: T, previous?: ListNode<T>) {
    this.payload = payload;
    if (previous) {
      previous.next = this;
    } else {
      this.next = null;
    }
  }
}
