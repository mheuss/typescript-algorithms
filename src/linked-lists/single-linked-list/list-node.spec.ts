import { ListNode } from './list-node';

describe('List Node unit test', () => {
  it('Should allow for creation of numerous node items', () => {
    const node = new ListNode<number>(5);
    const secondNode = new ListNode<number>(15, node);
    expect(node).toMatchSnapshot();
    expect(secondNode).toMatchSnapshot();
  });
});
