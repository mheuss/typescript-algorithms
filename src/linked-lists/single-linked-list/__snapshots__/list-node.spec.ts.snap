// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Node unit test Should allow for creation of numerous node items 1`] = `
ListNode {
  "next": ListNode {
    "next": null,
    "payload": 15,
  },
  "payload": 5,
}
`;

exports[`List Node unit test Should allow for creation of numerous node items 2`] = `
ListNode {
  "next": null,
  "payload": 15,
}
`;
